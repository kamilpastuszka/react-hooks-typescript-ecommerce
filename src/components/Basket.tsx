import React, { useContext } from "react";
import { Context } from "../context/context";
import BasketItem from "./BasketItem";
import styled from "styled-components";

export default function Basket() {
  const { state, dispatch } = useContext(Context);

  const removeProduct = (item: { id: number; name: string; price: number }) => {
    dispatch({ type: "REMOVE_PRODUCT", payload: item });
  };

  const SelectedItems = state.map(item => (
    <BasketItem
      key={item.id}
      name={item.name}
      price={item.price}
      removeProduct={() => removeProduct(item)}
    />
  ));

  const BasketContent =
    state.length >= 1 ? (
      SelectedItems
    ) : (
      <EmptyBasket>your basket is empty </EmptyBasket>
    );

  return (
    <table className="table mt-5">
      <tbody>{BasketContent}</tbody>
    </table>
  );
}

const EmptyBasket = styled.div`
  font-size: 2em;
  text-align: center;
`;
