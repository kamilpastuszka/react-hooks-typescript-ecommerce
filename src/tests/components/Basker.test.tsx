import React from "react";
import { shallow } from "enzyme";
import Basket from "../../components/Basket";
import toJson from "enzyme-to-json";

test("It should render Basket correctly", () => {
  const wrapper = shallow(<Basket />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
