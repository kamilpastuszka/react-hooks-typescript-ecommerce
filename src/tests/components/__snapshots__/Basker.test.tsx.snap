// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Basket correctly 1`] = `
<table
  className="table mt-5"
>
  <tbody>
    <BasketItem
      key="0"
      name=""
      price={0}
      removeProduct={[Function]}
    />
  </tbody>
</table>
`;
